internal/util/prometheus.go

```diff
@@ -0,0 +1,55 @@
+package util
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"sigs.k8s.io/controller-runtime/pkg/metrics"
+)
+
+type ControllerFunctionalPrometheusMetrics struct {
```

Comment:
Since it's idiomatic for Go to have simple type names - much simpler than for example Java:

maybe just call it `ControllerMetrics`?

Reply:
Want to high light they are controller business logic metrics, comparing to the metrics provided by controller-runtime. 
Thus, think this struct name makes more sense.

Reply:
I understand (and agree with) the motivation. But this name is way too elaborate to be convenient to use, by Go standards (comparing to other languages like Java that promote such long names).

Can you think of a short name that captures that meaning? For example `FunctionalMetrics`?

Reply:
how about "CtrlFuncMetrics" ?

Reply:
changed https://github.com/Kong/kubernetes-ingress-controller/pull/1622/commits/84df128254b7b56653f176a87b6d3efa470dd2c8